src/Control/Monad/ST.purs

```diff
@@ -62,3 +62,8 @@ foreign import runST
 -- | parentheses instead.
 pureST :: forall a. (forall h. Eff (st :: ST h) a) -> a
 pureST st = runPure (runST st)
+
+-- | Safely coerce a locally mutable `STRef h a` to a globally mutable `Ref a`
+-- | discarding the phantom type `h`.
+toRef :: forall a h. (STRef h a) -> (Ref a)
```

Comment:
I'll need it to unwrap `STRef` whether or not it's exported. It could also be moved to an `Unsafe` module.